pkg/apis/core/validation/validation.go

```diff
@@ -5185,9 +5260,11 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 	var newContainers []core.Container
 	for ix, container := range mungedPodSpec.Containers {
 		container.Image = oldPod.Spec.Containers[ix].Image // +k8s:verify-mutation:reason=clone
-		// When the feature-gate is turned off, any new requests attempting to update CPU or memory
-		// resource values will result in validation failure.
-		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		// CPU and memory resource updates will result in validation failures in two cases:
+		// 1. When the feature-gate is disabled
+		// 2. When using PodLevelResources, as it is incompatible with InPlacePodVerticalScaling
+		if opts.InPlacePodVerticalScalingEnabled &&
+			!(isPodLevelResourcesSet(oldPod) || isPodLevelResourcesSet(newPod)) {
```

Comment:
`isPodLevelResourcesSet` checks two things: 1) whether the feature PodLevelResource sis enabled and 2) whether pod level resources are set. 

Q: Is it possible to have a case where pod level resources are set but the feature gate is disabled?

Reply:
Yeah, I don't think you want to check the gate here.  I think simply checking if there are pod resources set AT ALL means that resize should fail.  Beware the `resources: {}` case - is that "set" or "not set" ?